### PR TITLE
Added the gravity compensation feature to the Aloha stack

### PR DESCRIPTION
**Pull request by Shiming-Liang**
Major changes:
- A client class in the `robot_utils.py` to enable/disable gravity compensation
- New entries in the launch file to bring up the gravity compensation nodes
- A `leader_motor_specs.yaml` for the gravity compensation nodes
- Slight changes to the `dual_side_teleop.py` and the `sleep.py` scripts to switch between torque on/off and enable/disable gravity compensation

### CODE REVIEW

File: config/leader_motor_specs.yaml
Comment (lukeschmitt-tr) [inline config/leader_motor_specs.yaml:1]: Let's make two files - leader_left and leader_right. We have seen how two different robots of the same model can act slightly differently — we might want to tune each arm individually.

File: scripts/dual_side_teleop.py [hunk 1]
```diff
@@ -12,8 +14,9 @@
     get_arm_gripper_positions,
     move_arms,
     move_grippers,
-    torque_off,
     torque_on,
+    torque_off,
+    enable_gravity_compensation,
```
Comment (lukeschmitt-tr) [inline scripts/dual_side_teleop.py]: alphabetize

File: scripts/dual_side_teleop.py [hunk 2]
```diff
@@ -154,4 +164,10 @@ def main() -> None:
 
 
 if __name__ == '__main__':
-    main()
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--gravity_compensation',
```
Comment (lukeschmitt-tr) [inline scripts/dual_side_teleop.py]:
```suggestion
        '-g', '--gravity_compensation',
```

File: scripts/dual_side_teleop.py [hunk 3]
```diff
@@ -154,4 +164,10 @@ def main() -> None:
...
+        action='store_true',
+        help='If set, gravity compensation will be enabled for the leader robots.',
```
Comment (lukeschmitt-tr) [inline scripts/dual_side_teleop.py]: If set, gravity compensation will be enabled for the leader robots when teleop starts.

File: scripts/record_episodes.py [hunk 4]
```diff
@@ -28,6 +28,8 @@
     Recorder,
     torque_off,
     torque_on,
+    enable_gravity_compensation,
+    disable_gravity_compensation,
```
Comment (lukeschmitt-tr) [inline scripts/record_episodes.py]: alphabetize

File: aloha/robot_utils.py [hunk 5]
```diff
@@ -293,3 +296,34 @@ def postprocess_base_action(base_action):
     linear_vel, angular_vel = base_action
     angular_vel *= 0.9
     return np.array([linear_vel, angular_vel])
+
+
+class GravityCompensationClient(Node):
```
Comment (lukeschmitt-tr) [inline aloha/robot_utils.py]: Refactor this to xs_modules as per our discussion offline

File: scripts/record_episodes.py [hunk 6]
```diff
@@ -395,4 +410,9 @@ def debug():
             ' a joystick controller or some other manual method.'
         ),
     )
+    parser.add_argument(
+        '--gravity_compensation',
```
Comment (lukeschmitt-tr) [inline scripts/record_episodes.py]:
```suggestion
        '-g','--gravity_compensation',
```

File: scripts/record_episodes.py [hunk 7]
```diff
@@ -395,4 +410,9 @@ def debug():
...
+        action='store_true',
+        help='If set, gravity compensation will be enabled for the leader robots.',
```
Comment (lukeschmitt-tr) [inline scripts/record_episodes.py]: If set, gravity compensation will be enabled for the leader robots when teleop starts.

File: aloha/robot_utils.py [hunk 8]
```diff
@@ -293,3 +296,34 @@ def postprocess_base_action(base_action):
...
+def enable_gravity_compensation(bot: InterbotixManipulatorXS):
+    client = GravityCompensationClient(bot.core.robot_name)
+    client.enable()
+    client.destroy_node()
+
+
+def disable_gravity_compensation(bot: InterbotixManipulatorXS):
+    client = GravityCompensationClient(bot.core.robot_name)
+    client.disable()
+    client.destroy_node()
```
Comment (lukeschmitt-tr) [inline aloha/robot_utils.py]: These simple calls can stay

**Review by lukeschmitt-tr — CHANGES_REQUESTED** [review]

File: scripts/auto_record.sh [hunk 9]
```diff
@@ -1,17 +1,17 @@
 ROS_DISTRO=humble
 
-VENV_ACTIVATE_PATH=$HOME/act/bin/activate
+# VENV_ACTIVATE_PATH=$HOME/act/bin/activate
```
Comment (lukeschmitt-tr) [inline scripts/auto_record.sh]: Remove commented lines if not necessary to be there

File: scripts/auto_record.sh [hunk 10]
```diff
@@ -31,7 +31,7 @@ echo "Task: $1"
 for (( i=0; i<$2; i++ ))
 do
   echo "Starting episode $i"
-  python3 "$RECORD_EPISODES" --task "$1"
+  python3 "$RECORD_EPISODES" --task "$1" $3 $4
```
Comment (lukeschmitt-tr) [inline scripts/auto_record.sh]: Put args in double quotes

```suggestion
  python3 "$RECORD_EPISODES" --task "$1" "$3" "$4"
```

**Review by lukeschmitt-tr — CHANGES_REQUESTED** [review]

File: scripts/auto_record.sh (hunk 10 above)
Comment (Shiming-Liang) [reply scripts/auto_record.sh]: The python script will complaint about getting `python3 record_episodes.py --task "xxx" "-g"`. But we can remove the "" for $1 to keep the style consistent. What do you think?

Comment (lukeschmitt-tr) [reply scripts/auto_record.sh]: Sure, that works for me.

Comment (Shiming-Liang) [reply scripts/auto_record.sh]: I looked into this more. It was the optional argument that caused the problem. We are getting `python3 record_episodes.py --task "xxx" "-g" ""`, where the last one is a empty str not defined in the arg parser. Either way, not using the quotes seems to be a good solution.

**Review by lukeschmitt-tr — APPROVED** [review]

**Review by shantanuparab-tr — APPROVED** [review]

**Pull request merged**